### ISSUE TITLE
Check whether the cite detector is finding good citations

### ISSUE THREAD

**Issue by lmullen** [lmullen/legal-modernism#67]
@kfunk074 asked for the citations we've found so far for these treatises.

- F0103272617
- F0151947538
- F0105827925
- F0103761500
- F0102826781

**Comment by lmullen** [comment]
The citations detected are attached. It's best to think about the information here as having two parts.

1. The citation detected (boils down to the `raw` field). The citations detected are not entirely final, but reasonably close.
2. The link between the citation and the case (the `case` field contains the CAP case ID). This was done quick and dirty, without any guessing for page numbers and so forth. So treat this as temporary data.

If there is a `reporter_abbr` field that looks like a reporter (i.e. not obviously junk) and there is not an entry in `reporter_cap`, then we want to add that to the whitelist.

If there is an entry in the `reporter_`cap` field (and thus in the `cleaner_cite` field) but not in the `case` field it would be helpful to know the categories of problems that are causing that. I.e., is the volume number wrong? The page number? Is the page number wrong in the actual text? Or is it an OCR error. Stuff like that.



[F0102826781.csv](https://github.com/lmullen/legal-modernism/files/9340041/F0102826781.csv)
[F0103272617.csv](https://github.com/lmullen/legal-modernism/files/9340043/F0103272617.csv)
[F0103761500.csv](https://github.com/lmullen/legal-modernism/files/9340042/F0103761500.csv)
[F0105827925.csv](https://github.com/lmullen/legal-modernism/files/9340045/F0105827925.csv)
[F0151947538.csv](https://github.com/lmullen/legal-modernism/files/9340044/F0151947538.csv)

**Comment by lmullen** [comment]
That is a bit strange. Part of it can be explained by the fact that there are reporters without whitelists. But not all, and not that much. I did something janky to write the query quickly. Let me try again a different way.

**Comment by lmullen** [comment]
Ha. Stupid bug.

**Comment by lmullen** [comment]
Here are twenty sample "citations" for each of the reporters to be whitelisted. The reporters are ordered by frequency, so best to start at the top and work down. Each "citation" has a link to a treatise in MOML, along with the page number of the citation. To make this file size manageable, it is only "reporters" that appear at least 1000 times, not 100 times as in the treatise-by-treatise files above.


[Examples of citations to reporterts not whitelisted.csv](https://github.com/lmullen/legal-modernism/files/9352350/Examples.of.citations.to.reporterts.not.whitelisted.csv)

**Comment by lmullen** [comment]
The following files supersede all those above. They have been updated to include a better list of what has already been whitelisted.

These are the top reporters still to be whitelisted, with sample citations for them.
[Top reporters not yet whitelisted.csv](https://github.com/lmullen/legal-modernism/files/9352783/Top.reporters.not.yet.whitelisted.csv)
[Sample citations for top reporters not yet whitelisted.csv](https://github.com/lmullen/legal-modernism/files/9352784/Sample.citations.for.top.reporters.not.yet.whitelisted.csv)

And these are the citations found for specific treatises, with the quick and temporary way of linking them to CAP cases.
[F0102826781.csv](https://github.com/lmullen/legal-modernism/files/9352785/F0102826781.csv)
[F0103272617.csv](https://github.com/lmullen/legal-modernism/files/9352786/F0103272617.csv)
[F0103761500.csv](https://github.com/lmullen/legal-modernism/files/9352787/F0103761500.csv)
[F0105827925.csv](https://github.com/lmullen/legal-modernism/files/9352788/F0105827925.csv)
[F0151947538.csv](https://github.com/lmullen/legal-modernism/files/9352789/F0151947538.csv)

**Comment by kfunk074** [comment]
RA has checked F010327617. Out of 166 citations on the page, we had 8 errors and 7 missing citations, for an overall accuracy of 91% (88% if we ignore statutory cites). Not bad. Only one of the errors was a number error. All the others were text OCR problems that will be resolved on our next round of whitelisting.

The only systemic issue is that we consistently missed citations to Howell's Special Term Reports, which look like this: **1 How. Sp. T. Rep. 114**. I checked the OCR on the MOML pages and it's flawless. We _did_ pick up citations to the reporter when there was an OCR error, for instance **How. Sp. T. Rep** (no terminal period) and **How. Sp.T. Rep.** (missing internal space) are both in the detected citations list. Is there an alpha character limit or something that's excluding the full cites?

**Comment by kfunk074** [comment]
Slight amendment: 3 cites were missed because the volume number **1** was OCR'd as **I** or **]**. Not worth addressing right now, but we can keep in mind that these kinds of errors happen.

**Comment by lmullen** [comment]
As you surmised, that abbreviation is 16 characters long, but the regex limited the abbreviations to 15 characters. I have made a change and added a test in https://github.com/lmullen/legal-modernism/commit/914bbeb2766f43066632460ebb4e0ae178498c7e

**Comment by kfunk074** [comment]
RA has spot checked the other four treatises. Overall, we have an initial accuracy rate of 75%. That will improve somewhat with the next round of whitelisting. A couple of the treatises just have really bad OCR on the numbers and there's nothing our current approach can do about it. Column breaks in footnotes threw a lot of things off, too. The good news is that the treatise with the most column break errors also used parallel cites abundantly, so if we're just trying to draw one edge between a treatise and a cited case, we probably have an average of four different chances to get it right between the parallel cites and the table of authorities.

Only systemic issue that arose is we've identified some more single-volume reporters. My recollection is you're keeping a separate library of those. If so, these should be added: Baldw., Comst., Cro. Car., Hob., Palm., Peake's Cas. All except the first appear to be UK reporters.

**Comment by kfunk074** [comment]
[reporters_citation_to_cap_whitelist-4.csv](https://github.com/lmullen/legal-modernism/files/9445009/reporters_citation_to_cap_whitelist-4.csv)

Update to the whitelist, covering a little over 1,000 entries and 5 million "cites." Let me know if there's a better format, but this is the format used for the previous whitelisting efforts.

**Comment by lmullen** [comment]
@kfunk074 I've added this to the database.

I'm not sure why, but about 50 of the entries were duplicated. I cleaned it up, and it's unlikely to happen again, so I wouldn't worry about it.

**Comment by kfunk074** [comment]
Another 1,000 entries to update the whitelist, hopefully with few duplicates, but there are probably some. This should cover all reporters cited more than 5,000 times (there may have been a troubling entry or two we left for further investigation).

[reporters_citation_to_cap_whitelist-5.csv](https://github.com/lmullen/legal-modernism/files/9575868/reporters_citation_to_cap_whitelist-5.csv)

**Comment by kfunk074** [comment]
At this point, to avoid duplicates, we should probably start with a fresh list of non-yet-whitelisted reporters cited more than 500 times. And the example entries have been really helpful, so we should provide an updated sheet of 10 random examples per each reporter for Sean as well (can be limited to more than 750 or 1000 cites if 500 is too large).

**Comment by kfunk074** [comment]
To collate the above into a task list:

- [ ] Update whitelist with [reporters_citation_to_cap_whitelist-5.csv](https://github.com/lmullen/legal-modernism/files/9576077/reporters_citation_to_cap_whitelist-5.csv)
- [ ] Run a new Top Reporters Not Yet Whitelisted for reporters cited over 500 times 
- [ ] Run a new Examples of Reporters Not Yet Whitelisted for reporters cited over 500 (or 750 or 1000) times 
- [ ] Pull the list of citations and their CAP matches for five random treatises for spot checking
- [ ] Distribute whitelisting sheets to Sean and spot checking sheets to Conor
